code/datums/riding.dm

```diff
@@ -34,16 +32,29 @@
 /datum/riding/proc/force_dismount(mob/living/M)
 	ridden.unbuckle_mob(M)
 
-//Override this to set your vehicle's various pixel offsets
-//if they differ between directions, otherwise use the
-//generic variables
 /datum/riding/proc/handle_vehicle_offsets()
+	var/ridden_dir = "[ridden.dir]"
+	var/passindex = 0
 	if(ridden.has_buckled_mobs())
 		for(var/m in ridden.buckled_mobs)
+			passindex++
 			var/mob/living/buckled_mob = m
-			buckled_mob.setDir(ridden.dir)
-			buckled_mob.pixel_x = generic_pixel_x
-			buckled_mob.pixel_y = generic_pixel_y
+			var/list/offsets = get_offsets(passindex)
+			dir_loop:
+				for(var/offsetdir in offsets)
+					if(offsetdir == ridden_dir)
+						var/list/diroffsets = offsets[offsetdir]
+						buckled_mob.pixel_x = diroffsets[1]
+						if(diroffsets.len == 2)
+							buckled_mob.pixel_y = diroffsets[2]
+						if(diroffsets.len == 3)
+							buckled_mob.layer = diroffsets[3]
+						break dir_loop
+
+
+//Override this to set your vehicle's various pixel offsets
+/datum/riding/proc/get_offsets(pass_index) // list(dir = x, y, layer)
+	return list("[NORTH]" = list(0, 0), "[SOUTH]" = list(0, 0), "[EAST]" = list(0, 0), "[WEST]" = list(0, 0))
```

Comment:
The defines are numbers, which forces a specific order, which would require an explanatory comment.
We pretty much always ```dir2text()``` or ```"[]"``` these defines.